tst/EnergyPlus/unit/EMSManager.unit.cc

```diff
@@ -2604,3 +2604,235 @@ TEST_F(EnergyPlusFixture, EMSManager_TrendValue_to_Actuator)
+
+        "EnergyManagementSystem:Sensor,",
+        "  ScheduleConstant_Sensor,                !- Name",
+        "  ScheduleConstant,                       !- Output:Variable or Output:Meter Index Key Name",
+        "  Schedule Value;                         !- Output:Variable or Output:Meter Name",
+
+        "EnergyManagementSystem:Program,",
+        "  sch_test,                               !- Name",
+        "  SET DummyVal = ScheduleConstant_Sensor; !- Program Line 1",
+
+        "EnergyManagementSystem:ProgramCallingManager,",
+        "  sch_test_pcm,                           !- Name",
+        "  BeginTimestepBeforePredictor,           !- EnergyPlus Model Calling Point",
+        "  sch_test;                               !- Program Name 1",
+
+        "Output:Variable,*,Schedule Value,Timestep;",
+    });
+
+    ASSERT_TRUE(process_idf(idf_objects));
+
+    constexpr int numBuiltInErlVars = 27; // 27 ErlVariables are constant and built-in variables
+    constexpr std::array<std::string_view, numBuiltInErlVars> builtInErlVarNames = {
+        "NULL",
+        "FALSE",
+        "TRUE",
+        "OFF",
+        "ON",
+        "PI",
+        "TIMESTEPSPERHOUR",
+        "YEAR",
+        "CALENDARYEAR",
+        "MONTH",
+        "DAYOFMONTH",
+        "DAYOFWEEK",
+        "DAYOFYEAR",
+        "HOUR",
+        "TIMESTEPNUM",
+        "MINUTE",
+        "HOLIDAY",
+        "DAYLIGHTSAVINGS",
+        "CURRENTTIME",
+        "SUNISUP",
+        "ISRAINING",
+        "SYSTEMTIMESTEP",
+        "ZONETIMESTEP",
+        "CURRENTENVIRONMENT",
+        "ACTUALDATEANDTIME",
+        "ACTUALTIME",
+        "WARMUPFLAG",
+    };
+    constexpr int expectedUserErlVarNums = 2;
+
+    // Apparently this needs to be called before GetScheduleData, i.e., init_state()
+    EMSManager::CheckIfAnyEMS(*state); // get EMS input
+    EXPECT_TRUE(state->dataGlobal->AnyEnergyManagementSystemInModel);
+
+    state->dataGlobal->TimeStepZone = 0.25;
+    state->dataGlobal->TimeStepsInHour = 4;    // must initialize this to get schedules initialized
+    state->dataGlobal->MinutesInTimeStep = 15; // must initialize this to get schedules initialized
+    state->dataGlobal->TimeStepZone = 0.25;
+    state->dataHVACGlobal->TimeStepSys = 0.25;
+    state->dataGlobal->TimeStepZoneSec = state->dataGlobal->TimeStepZone * Constant::rSecsInHour;
+    state->init_state(*state);
+
+    OutputProcessor::SetupTimePointers(
+        *state, OutputProcessor::TimeStepType::Zone, state->dataGlobal->TimeStepZone); // Set up Time pointer for HB/Zone Simulation
+    OutputProcessor::SetupTimePointers(*state, OutputProcessor::TimeStepType::System, state->dataHVACGlobal->TimeStepSys);
+
+    // Read EMS first, it should NOT find the Schedule Value in the output processor.
+    bool anyEMSRan = false;
+    EMSManager::ManageEMS(*state, EMSManager::EMSCallFrom::BeginZoneTimestepBeforeInitHeatBalance, anyEMSRan, ObjexxFCL::Optional_int_const());
+    EXPECT_FALSE(anyEMSRan);
```

Comment:
First call to ManageEMS, I have to ensure I don't find the output variable yet, or CheckedOkay is set and the .sched is never resolved in ProcessEMSInput (don't ask me why it does it in two passes)